### DISCUSSION TITLE
Form component Datepicker. Add option to add clear input button

### DISCUSSION THREAD

**Discussion by CodeTechNL** [filamentphp/filament#10643] [Ideas]
When you have a optional date field, it's not user friendly to clear the input data. A small X button or 'remove' button would be handy

**Comment by MuhamadMwaas** [comment]
up

**Comment by carvemerson** [comment]
A little late but I hope it can help other devs. You can archive this by using the suffixAction and a custom action

```php
Forms\Components\DatePicker::make('expires_at')
        ->native(false)
        ->suffixAction(
            Forms\Components\Actions\Action::make('clear')
                ->action(fn(Forms\Set $set) => $set('expires_at', null))
                ->tooltip('Clear Expiration Date')
                ->icon('heroicon-o-x-mark')
                ->label('Clear'))
```

Then you will see:

![image](https://github.com/user-attachments/assets/34ca2a9d-dd67-41d3-a9a9-b316dfd13927)

**Reply by CodeTechNL** [reply]
That makes a extra request for only clearing a input which can be done by JS. As a tmp solution you could use indeed a action.

**Reply by puzzledmonkey** [reply]
btw, at least in filament 5, you can use `actionJs("\$set('expires_at', null)")` instead of the `action(...)` call, which works instantly and avoids the extra network call. still, i agree this should be a native feature :+1:

**Comment by ariaieboy** [comment]
@danharrin 
We have something similar to this for select inputs it's a nice feature to have for all form inputs. a clear button so clear the state to default or placeholder.

**Comment by mattlanham** [comment]
+1
